components/channel_header_dropdown/menu_items/toggle_mute_channel/toggle_mute_channel.tsx

```diff
@@ -65,13 +64,22 @@ export default class MenuItemToggleMuteChannel extends React.PureComponent<Props
         const {
             id,
             isMuted,
+            channel,
         } = this.props;
 
         let text;
-        if (isMuted) {
-            text = localizeMessage('channel_header.unmute', 'Unmute Channel');
+        if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
+            if (isMuted) {
+                text = localizeMessage('channel_header.unmuteConversation', 'Unmute Conversation');
+            } else {
+                text = localizeMessage('channel_header.muteConversation', 'Mute Conversation');
+            }
         } else {
-            text = localizeMessage('channel_header.mute', 'Mute Channel');
+            if (isMuted) {
+                text = localizeMessage('channel_header.unmute', 'Unmute Channel');
+            } else {
+                text = localizeMessage('channel_header.mute', 'Mute Channel');
+            }
```

Comment:
We should avoid nesting `if/else` statements as much as possible. I can see that this is not easily done here, but we could at least simplify the assignment by using a ternary operator

```typescript
        if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
            text = isMuted ? 
                localizeMessage('channel_header.unmuteConversation', 'Unmute Conversation') :
                localizeMessage('channel_header.muteConversation', 'Mute Conversation');
        } else {
            text = isMuted ?
                localizeMessage('channel_header.unmute', 'Unmute Channel') :
                localizeMessage('channel_header.mute', 'Mute Channel');
        }
```

Reply:
Yeah will do that :+1: